thavalon-server/src/notifications/account.rs

```diff
@@ -0,0 +1,42 @@
+//! Module containing account-related notification code.
+
+use super::{send_email, NotificationError};
+use crate::database::accounts;
+use chrono::{Duration, Utc};
+use rand::{distributions::Alphanumeric, Rng};
+use std::iter;
+
+const EXPIRATION_DAYS: i64 = 3;
+const EMAIL_BASE_PATH: &str = "http://localhost:8001/api/verify_email/";
+
+/// Sends an email verification email to the client and adds the verification
+/// code to the database.
+///
+/// # Arguments
+///
+/// * `email` - The email to send to the client
+///
+/// # Returns
+///
+/// * Empty type on success, `NotificationError` on failure.
+pub async fn send_email_verification(email: &String) -> Result<(), NotificationError> {
+    log::info!("Sending a verification email for a new account.");
+    let code: String = iter::repeat(())
+        .map(|()| rand::thread_rng().sample(Alphanumeric))
```

Comment:
This'll look up the thread-local value every time - I'm not sure quite how expensive that is, but it might be worth pulling the `thread_rng` call out of the loop

Reply:
Due to the below `await`s, I had to put the rng declaration in its own block. Should be cached locally now though.